tests/ts_simple_recompile_ddl.erl

```diff
@@ -0,0 +1,87 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2015 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+
+-module(ts_simple_recompile_ddl).
+
+-behavior(riak_test).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-export([confirm/0]).
+-define(TABLE, ?MODULE).
+-define(KV_MODULE, riak_kv_compile_tab).
+
+confirm() ->
+    {Cluster, _Conn} = ts_util:cluster_and_connect(single),
+    Node = hd(Cluster),
+    lists:foreach(
+        fun(Table) ->
+            DDL = create_table_sql(Table),
+            ts_util:create_and_activate_bucket_type(Cluster, DDL, Table)
+        end, test_tables()),
+    rt:stop(Node),
+    simulate_old_dets_entries(),
+    rt:start(Node),
+    verify_resulting_dets_entries(),
+    pass.
+
+open_dets() ->
+    FileDir = rtdev:riak_data(1),
+    FilePath = filename:join(FileDir, [?KV_MODULE, ".dets"]),
+    {ok, ?TABLE} = dets:open_file(?TABLE, [{type, set}, {repair, force}, {file, FilePath}]).
+
+simulate_old_dets_entries() ->
+    open_dets(),
+    Pid = spawn_link(fun() -> ok end),
+    Pid2 = spawn_link(fun() -> ok end),
+    Pid3 = spawn_link(fun() -> ok end),
+    Table1DDL = sql_to_ddl(create_table_sql("Table1")),
+    Table2DDL = sql_to_ddl(create_table_sql("Table2")),
+    Table3DDL = sql_to_ddl(create_table_sql("Table3")),
+    ok = dets:insert(?TABLE, {<<"Table1">>, Table1DDL, Pid, compiled}),
+    ok = dets:insert(?TABLE, {<<"Table2">>, 1, Table2DDL, Pid2, compiled}),
+    ok = dets:insert(?TABLE, {<<"Table3">>, 1, Table3DDL, Pid3, compiling}),
```

Comment:
That `compiling` atom you are inserting last, isn't it going to make the `verify_resulting_dets_entries` fail?

Generally, some comments here would help reviewers to understand the underlying logic of the test. As it is, in `verify_resulting_dets_entries` you are reading the entries you have just inserted (in `simulate_old_dets_entries`), which is bound to succeed.

Aha, got it. The `'compiling'` will be turned to `'compiled'` in the process of node going up. I'm smart, no?